hoodie-common/src/main/java/com/uber/hoodie/common/util/FSUtils.java

```diff
@@ -112,29 +119,52 @@ public static long getFileSize(FileSystem fs, Path path) throws IOException {
         return fs.listStatus(path)[0].getLen();
     }
 
-    public static String globAllFiles(String basePath) {
-        return String.format("%s/*/*/*/*", basePath);
-    }
-
-    // TODO (weiy): rename the function for better readability
     public static String getFileId(String fullFileName) {
         return fullFileName.split("_")[0];
     }
 
+
     /**
-     * Obtain all the partition paths, that are present in this table.
+     * Gets all partition paths assuming date partitioning (year, month, day) three levels down.
      */
-    public static List<String> getAllPartitionPaths(FileSystem fs, String basePath)
-        throws IOException {
-        List<String> partitionsToClean = new ArrayList<>();
-        // TODO(vc): For now, assume partitions are two levels down from base path.
+    public static List<String> getAllFoldersThreeLevelsDown(FileSystem fs, String basePath) throws IOException {
+        List<String> datePartitions = new ArrayList<>();
         FileStatus[] folders = fs.globStatus(new Path(basePath + "/*/*/*"));
         for (FileStatus status : folders) {
             Path path = status.getPath();
-            partitionsToClean.add(String.format("%s/%s/%s", path.getParent().getParent().getName(),
-                path.getParent().getName(), path.getName()));
+            datePartitions.add(String.format("%s/%s/%s", path.getParent().getParent().getName(),
+                    path.getParent().getName(), path.getName()));
         }
-        return partitionsToClean;
+        return datePartitions;
+    }
+
+    /**
+     * Obtain all the partition paths, that are present in this table, denoted by presence of {@link
+     * com.uber.hoodie.common.model.HoodiePartitionMetadata#HOODIE_PARTITION_METAFILE}
+     */
+    public static List<String> getAllFoldersWithPartitionMetaFile(FileSystem fs, String basePathStr)
+            throws IOException {
+        List<String> partitions = new ArrayList<>();
+        Path basePath = new Path(basePathStr);
+        RemoteIterator<LocatedFileStatus> allFiles = fs.listFiles(new Path(basePathStr), true);
```

Comment:
is `**` a special glob for matching files at any levels of subdirectories? If yes, we could do that. 

If no, then there is no good way since we don't know the depth of each partition.. it can be anything, and even vary across dataset.